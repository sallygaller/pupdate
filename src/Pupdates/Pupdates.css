.Pupdates {
    text-align: center;
    margin-left: 0px;
}

.Pupdates ul, li {
    list-style-type: none;
    list-style-position: inside;
    margin: 0;
    padding: 0; 
}

.Pupdates li {
    margin-left: 0px;
}

.Pupdates-button {
    margin: 0 auto;
    text-align: center;
}

.Pupdates section {
    margin-bottom: 100px;
}

.Pupdates a,
.Pupdates a:active,
.Pupdates a:focus,
.Pupdates a:visited {
  color: black;
  font-weight: bold;
  text-decoration: none;
}