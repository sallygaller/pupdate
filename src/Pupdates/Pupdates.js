import React from "react";
import Pupdate from "../Pupdate/Pupdate";
import { API_ENDPOINT } from "../config";
import TokenService from "../services/token-service";
import "./Pupdates.css";

class Pupdates extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      pupdateOrganized: [],
      pupdateAttending: [],
      error: null,
    };
  }

  componentDidMount() {
    Promise.all([
      // get pupdate rsvps for logged in user
      fetch(API_ENDPOINT + `/pupdate-rsvp/user`, {
        headers: {
          authorization: `bearer ${TokenService.getAuthToken()}`,
        },
      }),
      // get pupdates that user has organized
      fetch(API_ENDPOINT + `/pupdates/user`, {
        headers: {
          authorization: `bearer ${TokenService.getAuthToken()}`,
        },
      }),
    ])
      .then(([res1, res2]) => Promise.all([res1.json(), res2.json()]))
      .then(([responseData1, responseData2]) => {
        this.setState({
          pupdateOrganized: responseData2,
        });
        Promise.all(
          responseData1.map((pupdateRsvp) =>
            fetch(API_ENDPOINT + `/pupdates/${pupdateRsvp.pupdate}`, {
              headers: {
                authorization: `bearer ${TokenService.getAuthToken()}`,
              },
            })
          )
        )
          .then((responses) => {
            return Promise.all(
              responses.map((response) => {
                return response.json();
              })
            );
          })
          .then((data) => {
            console.log(data);
            this.setState({
              pupdateAttending: data,
            });
          });
      })
      .catch((error) => {
        console.error({ error });
      });
  }

  render() {
    return (
      <div className="Pupdates">
        <section>
          <h3>Pupdates I've organized:</h3>
          {this.state.pupdateOrganized.length === 0 ? (
            <p>You haven't organized a pupdate</p>
          ) : (
            <ul>
              {this.state.pupdateOrganized.map((pupdate) => (
                <li key={pupdate.id}>
                  <Pupdate pupdate={pupdate} userOrganized="true" />
                </li>
              ))}
            </ul>
          )}
          <h3>Pupdates I'm attending</h3>
          <div>
            <ul>
              {this.state.pupdateAttending.map((pupdate) => (
                <li key={pupdate.id}>
                  <Pupdate pupdate={pupdate} userAttending="true" />
                </li>
              ))}
            </ul>
          </div>
        </section>
      </div>
    );
  }
}

export default Pupdates;
