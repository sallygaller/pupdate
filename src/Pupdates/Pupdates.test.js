import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Pupdates from "./Pupdates";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <Pupdates />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer.create(<Pupdates name="Pupdates" />).toJSON();
  expect(tree).toMatchSnapshot();
});
