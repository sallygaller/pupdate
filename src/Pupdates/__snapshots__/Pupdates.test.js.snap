// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Pupdates"
>
  <h2>
    Pupdates
  </h2>
  <section>
    <div
      className="main"
    >
      <div
        className="section"
      >
        <div
          className="sectionhead"
          onClick={[Function]}
        >
          <h3>
            Pupdates I've Organized
          </h3>
        </div>
        <div
          className="page fadeOut"
        >
          <div
            className="article"
          >
             
            <p>
              No pupdates here! Why don't you create one?
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="section"
      >
        <div
          className="sectionhead"
          onClick={[Function]}
        >
          <h3>
            Pupdates I'm Attending
          </h3>
        </div>
        <div
          className="page fadeOut"
        >
          <div
            className="article"
          >
             
            <p>
              No pupdates here! Why don't you create one?
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="AvailablePupdates"
    >
      <section>
        <div>
          <div
            className="main"
          >
            <div
              className="section"
            >
              <div
                className="sectionhead"
                onClick={[Function]}
              >
                <h3>
                  Available Pupdates
                </h3>
              </div>
              <div
                className="page fadeOut"
              >
                <div
                  className="article"
                >
                   
                  <p>
                    No pupdates here! Why don't you create one?
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </section>
</div>
`;
